### ISSUE TITLE
homebase-indexer: Expand README

### ISSUE THREAD

**Issue by benefacto** [dOrgTech/homebase-app#736]
## Overview

The `homebase-indexer` project currently has a placeholder README. This issue proposes a comprehensive expansion of the README for better guidance.

### Comprehensive README Update
- **Goal:** To provide detailed, step-by-step instructions and project insights for new users and contributors.
- **Details:**
  - Introduction: Brief overview of `homebase-indexer`, its purpose, and how it integrates with DipDup.
  - Prerequisites: List required software and knowledge, like Python version, Docker, etc.
  - Installation and Setup: Detailed instructions on setting up the project environment.
  - Usage: How to use the indexer, with basic commands and examples.
  - Configuration: Guide on configuring the indexer, with references to the DipDup quickstart guide (https://docs.dipdup.io/quickstart).
  - Contribution Guidelines: Instructions for contributors on how to propose changes, report issues, etc.
  - Troubleshooting: Common issues and their solutions.
  - License: Information about the project's licensing.

## Expected Outcome

- A detailed README providing comprehensive guidance for setting up, using, and contributing to `homebase-indexer`.
- Enhanced user and contributor experience, potentially increasing the project's user base and active contributions.

## Additional Notes

- Ensure all instructions are tested and verified for accuracy.
- Consider linking to external resources for more in-depth explanations of specific technologies or concepts.

## References

- Current `homebase-indexer` project: https://github.com/dOrgTech/homebase-indexer
- DipDup Quickstart Documentation: https://docs.dipdup.io/quickstart

**Comment by benefacto** [comment]
Removed Docker Compose from scope as dipdup with the existing Dockerfile is a bit painful to get that setup without potentially breaking a Dockerfile that's working in production

**Comment by benefacto** [comment]
Actually adding Docker Compose is a bit painful with the existing Dockerfile and Dipdup setup. Given that the Dockerfile is already working in production, going to stick with just the Readme updates and Swagger addition for this scope.

**Comment by benefacto** [comment]
Docker Compose is actually already implemented on the _real_ main branch: https://github.com/dOrgTech/homebase-indexer/tree/v3-develop , just not merged into master (unclear what the purpose of the master branch is currently)

**Comment by benefacto** [comment]
Pull request is ready for review: https://github.com/dOrgTech/homebase-indexer/pull/37

**Comment by benefacto** [comment]
Team said I can merge this (may also want to merge to v2-develop)
